docs/content/concepts/assets/asset-observations.mdx

```diff
@@ -0,0 +1,99 @@
+---
+title: Asset Observations | Dagster
+description: Dagster provides functionality to record metadata about assets.
+---
+
+# Asset Observations
+
+An asset observation is an event that records metadata about a given asset.
+
+## Relevant APIs
+
+| Name                                                    | Description                                                          |
+| ------------------------------------------------------- | -------------------------------------------------------------------- |
+| <PyObject module="dagster" object="AssetObservation" /> | Dagster event indicating that an asset's metadata has been recorded. |
+| <PyObject module="dagster" object="AssetKey" />         | A unique identifier for a particular external asset                  |
+
+## Overview
+
+<PyObject module="dagster" object="AssetObservation" /> events are used to communicate
+to Dagster that metadata has been recorded about a given asset. Asset observation
+events can be yielded at runtime within ops and assets. An asset must be defined
+using the <PyObject object="asset" decorator /> decorator or have existing materializations
+in order for its observations to be displayed.
+
+## Yielding an AssetObservation from an Asset
+
+To make Dagster aware that we have recorded metadata about an asset, we can yield an <PyObject module="dagster" object="AssetObservation" /> event:
+
+```python file=/concepts/assets/observations.py startafter=start_observation_asset_marker_0 endbefore=end_observation_asset_marker_0
+from dagster import AssetObservation
+
+
+@asset
+def observation_asset():
+    df = read_df()
+    remote_storage_path = persist_to_storage(df)
+    yield AssetObservation(asset_key=AssetKey("observation_asset"), metadata={"num_rows": len(df)})
```

Comment:
This reminds me that we should add the ability to do `context.log_asset_observation` in the same way that users can now do `context.log_asset_materialization`.

Reply:
It's already been added [here](https://github.com/dagster-io/dagster/blob/25abc874d13decf8efdc01899f112aacced8d3e3/python_modules/dagster/dagster/core/execution/context/compute.py#L300)